woocommerce/woocommerce
# Optimize system status tests

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR optimizes the tests that check the response for the system status report from the REST API. Instead of fetching the same report many times just to check a single aspect of it, we now cache the result. Closes #38970.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. ensure checks pass
2. ensure [this manual CI run](https://github.com/woocommerce/woocommerce/actions/runs/5623194117) passes



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
